### PR TITLE
Complete test coverage for recorder.go of middleware package

### PR DESCRIPTION
**Pull request by Karthic-Hackintosh**
Checking the approach , want review

### CODE REVIEW

File: middleware/recorder_test.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+)
+
+func TestNewResponseRecorder(t *testing.T) {
+	w := httptest.NewRecorder()
+	recordRequest := NewResponseRecorder(w)
+	if !reflect.DeepEqual(recordRequest.ResponseWriter, w) {
```
Comment (abiosoft) [inline middleware/recorder_test.go]: `recordRequest.ResponseWriter == w` should be sufficient here.

**Comment by abiosoft** [comment]
Thanks a lot for this. If you are willing, you can test the `Write` and `WriteHeader` functions also.

**Comment by kraoadori** [comment]
@abiosoft  : yes , I'm writing the tests for those 2 functions also , but wanted to make sure that this test is right

**Comment by abiosoft** [comment]
Thanks once again :+1:

**Pull request merged**